src/controllers/rsoController.js

```diff
@@ -86,24 +94,92 @@ class RsoController extends ApiController {
       return next(new ApiError.InvalidPermissionForAction({ action: 'rso.create', userId: req.user.id }))
     }
 
-    let payload = _.merge(params, {
-      createdById: req.user.id
-    })
+    // Check to make sure that there are enough emails being
+    // sent to have five people in the RSO.
+    // Note: We only check for having at least 4 because the
+    // fifth user is assumed to be the user who is signed in.
+    if (params.memberEmails.length < 4) {
+      return next(new ApiError.NotEnoughMembersInRso(params))
+    }
 
-    db.Rso.create(payload)
-      .then((rso) => {
-        // Create a membership for the current user
-        // as an admin in the new RSO
-        req.user.addRso(rso, {
-          permissionLevel: permLevels.ADMIN,
-          universityId: params.universityId
-        }).then((instance) => {
-          res.json(rso)
-        })
-      }, (response) => {
-        console.log('Failed to create new RSO in university', response)
-        return next(new ApiError.NoRsoInUniversity({ action: 'rso#create', params: params, response: response }))
+    // Because of the above check, this is guaranteed to have
+    // at least four emails in it.
+    let userChecks = _.concat([],
+      db.User.findOne({ where: { email: params.memberEmails[0], inactiveAt: null } }),
+      db.User.findOne({ where: { email: params.memberEmails[1], inactiveAt: null } }),
+      db.User.findOne({ where: { email: params.memberEmails[2], inactiveAt: null } }),
+      db.User.findOne({ where: { email: params.memberEmails[3], inactiveAt: null } })
+    )
+
+    let usersToAdd = []
+    Promise.all(userChecks)
+      .then((results) => {
+        if (_.every(results, res => res)) {
+          usersToAdd = results
+
+          let checkingForUniMemb = _.concat([],
+            usersToAdd[0].isMemberOfUniversity(params.universityId),
+            usersToAdd[1].isMemberOfUniversity(params.universityId),
+            usersToAdd[2].isMemberOfUniversity(params.universityId),
+            usersToAdd[3].isMemberOfUniversity(params.universityId)
+          )
+
+          Promise.all(checkingForUniMemb)
+            .then((isMemb) => {
+              if (_.every(isMemb, r => r)) {
+                executeCreation(params)
+              } else {
+                const returnRes = _.zipObject(params.memberEmails, _.map(results, r => !!r))
```

Comment:
LOL! That is the international convert-to-boolean "operator". It's basically a really short-hand way of saying "return true if the object exists, otherwise return false".